packages/web-components/src/tooltip/tooltip.ts

```diff
@@ -0,0 +1,198 @@
+   */
+  private get anchorElement(): HTMLElement | null {
+    const rootNode = this.getRootNode();
+    return (rootNode instanceof ShadowRoot ? rootNode : document).getElementById(this.anchor ?? '');
+  }
+
+  /**
+   * Reference to the anchor positioning style element
+   * @internal
+   */
+  protected anchorPositioningStyleElement: HTMLStyleElement | null = null;
+
+  public constructor() {
+    super();
+    this.elementInternals.role = 'tooltip';
+  }
+
+  public connectedCallback(): void {
+    super.connectedCallback();
+    if (this.anchorElement) {
+      // @ts-expect-error - Baseline 2024
+      const anchorName = this.anchorElement.style.anchorName || `--${this.anchor}`;
+      this.anchorElement.setAttribute('aria-describedby', this.id);
+
+      if (SUPPORTS_ANCHOR_POSITIONING) {
+        // @ts-expect-error - Baseline 2024
+        this.anchorElement.style.anchorName = anchorName;
+        // @ts-expect-error - Baseline 2024
+        this.style.positionAnchor = anchorName;
+      } else {
+        // Provide style fallback for browsers that do not support anchor positioning
+        if (!this.anchorPositioningStyleElement) {
+          this.anchorPositioningStyleElement = document.createElement('style');
+          document.head.append(this.anchorPositioningStyleElement);
+        }
+
+        // Given a position with <direction>-<alignment> format, return the proper CSS properties
+        const [direction, alignment] = this.positioning?.split('-') ?? [];
+        let centerAlignment;
+
+        if (alignment === undefined) {
+          if (direction === 'above' || direction === 'below') {
+            centerAlignment = 'centerX';
+          }
+          if (direction === 'before' || direction === 'after') {
+            centerAlignment = 'centerY';
+          }
+        }
+
+        const directionCSS = {
+          above: `bottom: anchor(${anchorName} top);`,
+          below: `top: anchor(${anchorName} bottom);`,
+          before: `right: anchor(${anchorName} left);`,
+          after: `left: anchor(${anchorName} right);`,
+        }[direction];
+
+        const alignmentCSS = {
+          start: `left: anchor(${anchorName} left);`,
+          end: `right: anchor(${anchorName} right);`,
+          top: `top: anchor(${anchorName} top);`,
+          bottom: `bottom: anchor(${anchorName} bottom);`,
+          centerX: `left: anchor(${anchorName} center); translate: -50% 0;`,
+          centerY: `top: anchor(${anchorName} center); translate: 0 -50%;`,
+        }[alignment ?? centerAlignment];
+
+        this.anchorPositioningStyleElement.textContent = `
+          #${this.anchor} {
+            anchor-name: ${anchorName};
+          }
+          #${this.id} {
+            inset: unset;
+            ${directionCSS}
+            ${alignmentCSS}
+            position: absolute;
+          }
+        `;
+
+        // Dispatch a ready event for the anchor polyfill
+        document.dispatchEvent(new CustomEvent('anchor-polyfill'));
```

Comment:
This code handles the Anchor Positioning fallbacks (provided by the oddbird polyfill)

1. create a style tag if needed and append it to the head
2. convert the `positioning` attribute values into the proper direction and alignment css (unique ID for each tooltip element)
3. update the style tag in the head for the element.
4. dispatch a custom event to notify the polyfill it's okay to run

The last part needs the most feedback. The unique constraint is that the polyfill has to run after the CSS has been setup. You can see that in the `.storybook/preview-head.html`. I don't necessarily love it, but it works even when there are dozens of tooltips on the same page. I'm open to suggestions.